### PR TITLE
Backport 2.7: Update soon to be expired crl

### PR DESCRIPTION
**Pull request by RonEld**
## Description
Backport of #2417 to `mbedtls-2.7`

## Status
**READY**

### CODE REVIEW

**Comment by RonEld** [comment]
I have backported the bug fixes from #2417 to `mbedtls-2.7` in this PR.
However, since there were many conflicts in the certificates, I manually created the certificates again. This is why the certificates, and some of the test data is different than the original PR

@AndrzejKurek @k-stachowiak Please review

File: tests/data_files/Makefile [hunk 1]
```diff
@@ -48,6 +48,18 @@ test-ca-sha256.crt: $(test_ca_key_file_rsa) $(test_ca_config_file) test-ca.csr
 	$(OPENSSL) req -x509 -config $(test_ca_config_file) -key $(test_ca_key_file_rsa) -passin "pass:$(test_ca_pwd_rsa)" -set_serial 0 -days 3653 -sha256 -in test-ca.csr -out $@
```
Comment (AndrzejKurek) [inline tests/data_files/Makefile]: Don't we want the notbefore and notafter to be set for test-ca.crt, test-ca-sha1.crt and test-ca-sha256.crt?

Comment (RonEld) [reply tests/data_files/Makefile]: But we do.
In `mbedtls-2.7` the certificates are generated using openssl, as opposed to `mbedtls-2.16` which uses `cert_write`.
in openssl, the notbefore and notafter are generated with the `-days` parameter

**Comment by RonEld** [comment]
@k-stachowiak @AndrzejKurek I amended the last commit to fix CI failure.
It was only removing of an extra `;` character in `certs.c` line 203. This caused a failure in ARMCC5:
~~~
"certs.c", line 205: Error:  #381-D: extra ";" ignored
  const char mbedtls_test_srv_crt_rsa[]     =  
EST_SRV_CRT_RSA_SHA256;
                                                                      ^
certs.c: 0 warnings, 1 error
Makefile:164: recipe for target 'certs.o' failed
make[1]: *** [certs.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make: *** [lib] Error 2
Makefile:18: recipe for target 'lib' failed
^^^^build_armcc: build: ARM Compiler 5, make: command make CC=/usr/local/ARM_Compiler_5.06u3/bin//armcc AR=/usr/local/ARM_Compiler_5.06u3/bin//armar WARNING_CFLAGS=--strict --c99 lib -> 2^^^^
~~~
Please review

**Comment by RonEld** [comment]
CI failures in `compat.sh` when GnuTLs is used as server.
I will investigate

**Comment by RonEld** [comment]
I believe the failure with GnuTLS server is because the client certificate sent, `server1.crt`, issuer name is encoded with PrintableString, and the CA file Subject Name is encoded with UTF8-String.


In GnuTLS [``is_issuer function](https://gitlab.com/gnutls/gnutls/blob/gnutls_3_4_10/lib/x509/verify.c#L251), there is a raw buffer comparison, which should fail because of different encoding.

**Comment by RonEld** [comment]
@AndrzejKurek @k-stachowiak I have updated the certificates to fix the interop tests
Please review

**Comment by RonEld** [comment]
@AndrzejKurek @k-stachowiak I have rebased to resolve conflicts
Please review

**Comment by RonEld** [comment]
This PR is fully reviewed, so removing the "needs backports" label
cc @Patater

**Pull request merged**